docs/decisions/0025-dsp-pipeline-architecture.md

```diff
@@ -0,0 +1,524 @@
+│  │  - Waterfall (WebGL2/Canvas)                │  │              │
+│  │  - Spectrum Analyzer                        │  │              │
+│  │  - Constellation Diagram                    │  │              │
+│  └────────────────────────────────────────────┘  │              │
+│         ▲                                         │              │
+│         │ FFT results (transferable)             │              │
+└─────────┼─────────────────────────────────────────┼──────────────┘
+          │                                         │
+          │                                         │
+┌─────────┼─────────────────────────────────────────┼──────────────┐
+│         │                    DSP WORKER           │              │
+├─────────┼─────────────────────────────────────────┼──────────────┤
+│         │                                         │              │
+│         │                                         ▼              │
+│  ┌──────┴──────┐                      ┌──────────────────┐      │
+│  │ FFT Result  │                      │   Consumer       │      │
+│  │ (transfer)  │◀─────────────────────│ (read from SAB) │      │
+│  └─────────────┘                      └────────┬─────────┘      │
+│                                                 │                │
+│                                                 ▼                │
+│                                       ┌──────────────────┐       │
+│                                       │  DC Correction   │       │
+│                                       │  (WASM/SIMD)     │       │
+│                                       └────────┬─────────┘       │
+│                                                │                 │
+│                                                ▼                 │
+│                                       ┌──────────────────┐       │
+│                                       │   Windowing      │       │
+│                                       │  (Hann/Hamming)  │       │
+│                                       │  (WASM/SIMD)     │       │
+│                                       └────────┬─────────┘       │
+│                                                │                 │
+│                                                ▼                 │
+│                                       ┌──────────────────┐       │
+│                                       │   FFT            │       │
+│                                       │ (WASM Cooley-    │       │
+│                                       │  Tukey radix-2)  │       │
+│                                       └────────┬─────────┘       │
+│                                                │                 │
+│                                                ▼                 │
+│                                       ┌──────────────────┐       │
+│                                       │  Demodulation    │       │
+│                                       │  (AM/FM/SSB)     │       │
```

Comment:
Remember that we'll want an opportunity to have visibility into the DSP (so we can visualize steps of the pipeline independently) and will need to configure some settings from the UI (I don't see that pattern in this diagram)